### ISSUE TITLE
Invalid closure configuration methods lead to cryptic error messages

### ISSUE THREAD

**Issue by YaoWang0515** [gradle/gradle#14984]
When I used gradle-6.3-all in project's gradle-wrapper.properties,and configured the DSL incorrectly,the error message：
![image](https://user-images.githubusercontent.com/29978458/97249028-03757380-183e-11eb-9307-ec52d755b406.png)
![image](https://user-images.githubusercontent.com/29978458/97249061-125c2600-183e-11eb-8c89-4478cee04018.png)

but when using versions below 6.3,the error message：
![image](https://user-images.githubusercontent.com/29978458/97249146-3a4b8980-183e-11eb-8b2f-aed825119254.png)
![image](https://user-images.githubusercontent.com/29978458/97249165-47687880-183e-11eb-890e-20411162e385.png)

**Comment by blindpirate** [comment]
This doesn't seem like a Gradle issue. It seems that you're not applying AGP in the first screenshot. If you believe you're correct, a minimal reproducible sample would be great.

**Comment by YaoWang0515** [comment]
In fact,I have applied AGP in the first screenshot,I just didn't cut it out
![image](https://user-images.githubusercontent.com/29978458/97255564-df209380-184b-11eb-9b45-335ba4e8a517.png)

**Comment by blindpirate** [comment]
Can you please provide a sample project which reproduces this issue?

**Comment by YaoWang0515** [comment]
I just use Android Studio3.6.2 create a new project,and change the gradle-wrapper with gradle-6.3-all like the screenshot.
Then I change dsl 'compileSdkVersion'  incorrectly in build.gradle in app module

Sorry ,I don't know why the zip file can't be upload
![image](https://user-images.githubusercontent.com/29978458/97267232-ca9cc500-1864-11eb-9159-6a97c5513014.png)

**Comment by blindpirate** [comment]
Can you please upload to a github repo?

**Comment by YaoWang0515** [comment]
[MyApplication.zip](https://github.com/gradle/gradle/files/5443869/MyApplication.zip)

**Comment by blindpirate** [comment]
Thanks for the sample. I confirmed this issue. It seems like a regression to me. Probably caused by Groovy upgrade (6.2 is shipped with Groovy 2.5.8 and 6.3 is shipped with Groovy 2.5.10).

**Comment by blindpirate** [comment]
@gradle/core Did you see similar issues when upgrading Groovy?

**Comment by YaoWang0515** [comment]
Ok,Thank you for your confirmation and response!

**Comment by YaoWang0515** [comment]
The most common scenatio is a DSL configuration error.What else is not found.

**Comment by bmuskalla** [comment]
This can be reproduced with any closure-/action-taking method by introducing a typo.

```
plugins {
    id "java"
}

java {
    toolchains { // should be toolchains

    }
}
```

Error message:
```
No signature of method: build_5j61lan4ubea9vr3sx6340c0t.java() is applicable for argument types: (build_5j61lan4ubea9vr3sx6340c0t$_run_closure1) values: [build_5j61lan4ubea9vr3sx6340c0t$_run_closure1@2b6250f9]
  Possible solutions: wait(), any(), wait(long), any(groovy.lang.Closure), tap(groovy.lang.Closure), each(groovy.lang.Closure)
```

**Comment by ljacomet** [comment]
Fix will be released with Gradle 7.5
